go/api/options/zadd_options.go

```diff
@@ -0,0 +1,129 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package options
+
+import (
+	"errors"
+	"strconv"
+)
+
+/*
+ * Optional arguments to `Zadd` in [SortedSetCommands]
+ */
+type ZAddOptions struct {
+	/**
+	 * Defines conditions for updating or adding elements with {@link SortedSetBaseCommands#zadd}
+	 * command.
+	 */
+	conditionalChange ConditionalChange
+	/**
+	 * Specifies conditions for updating scores with zadd {@link SortedSetBaseCommands#zadd} command.
+	 */
+	updateOptions UpdateOptions
+	/**
+	 * Changes the return value from the number of new elements added to the total number of elements changed.
+	 */
+	changed bool
+	/**
+	 * If true, the zadd command will act like ZINCRBY.
+	 */
+	incr bool
+	/**
+	 * The increment value to use when incr is true.
+	 */
+	increment float64
+	/**
+	 * The member to add to.
+	 */
+	member string
+}
+
+func NewZaddOptionsBuilder() *ZAddOptions {
+	return &ZAddOptions{}
+}
+
+func (options *ZAddOptions) SetConditionalChange(c ConditionalChange) *ZAddOptions {
+	options.conditionalChange = c
+	return options
+}
+
+func (options *ZAddOptions) SetUpdateOptions(u UpdateOptions) *ZAddOptions {
+	options.updateOptions = u
+	return options
+}
+
+func (options *ZAddOptions) SetChanged(ch bool) (*ZAddOptions, error) {
+	if options.incr {
+		return nil, errors.New("changed cannot be set when incr is true")
+	}
+	options.changed = ch
+	return options, nil
+}
+
+func (options *ZAddOptions) SetIncr(incr bool, increment float64, member string) (*ZAddOptions, error) {
+	if options.changed {
+		return nil, errors.New("incr cannot be set when changed is true")
+	}
+	options.incr = incr
+	options.increment = increment
+	options.member = member
+	return options, nil
+}
+
+func (opts *ZAddOptions) ToArgs() ([]string, error) {
+	args := []string{}
+	var err error
+
+	if opts.conditionalChange != "" {
```

Comment:
We also need to validate that conditionalChange only has values `XX or NX` and throw an error if it's some other value.